src/test/fakeExtensionContext.ts

```diff
@@ -54,14 +56,32 @@ export class FakeExtensionContext implements ExtensionContext {
     public static async getNew(): Promise<FakeExtensionContext> {
         const ctx = new FakeExtensionContext()
         try {
-            ExtensionDisposableFiles.getInstance().dispose()
+            TestExtensionDisposableFiles.clearInstance()
         } catch {
-            await ExtensionDisposableFiles.initialize(ctx)
+            // Ignore.
         }
+        await TestExtensionDisposableFiles.initialize(ctx)
         return ctx
     }
 }
 
+export class TestExtensionDisposableFiles extends ExtensionDisposableFiles {
+    public static ORIGINAL_INSTANCE = ExtensionDisposableFiles.INSTANCE
+
+    public static clearInstance() {
+        const instance = ExtensionDisposableFiles.INSTANCE
+        ExtensionDisposableFiles.INSTANCE = undefined
+        if (instance) {
+            del.sync([instance.toolkitTempFolder], { force: true })
+            instance.dispose()
+        }
```

Comment:
Note that this logic was rearranged (mentioned in the commit message): `ExtensionDisposableFiles.INSTANCE=undefined` is done first, because 

1. `del.sync()` does file IO
2. the node scheduler looks for pending handlers to execute while waiting on IO
3. that cause a race: other async handlers may try to use `ExtensionDisposableFiles.getInstance()` which then throws an exception

Reply:
It may be worth adding some comments to the code around this (more visible than commit messages)